src/Runner.Worker/ExecutionContext.cs

```diff
@@ -650,6 +652,8 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
             // ActionsStepTelemetry
             ActionsStepsTelemetry = new List<ActionsStepTelemetry>();
 
+            JobTelemetry = new List<JobTelemetry>();
```

Comment:
should we pass around the `JobTelemetry` when we call `CreateChild()` so all ExecutionContext will share a single `List<JobTelemetry>`?

Reply:
we have a `GlobalContext` on the ExecutionContext to hold global info